internal/controller/k8s_resources.go

```diff
@@ -79,6 +83,7 @@ func convertObjectToMetadata(obj metav1.Object) metav1.ObjectMeta {
 // getObjectMetadataFromRuntimeObject safely extracts metadata from any Kubernetes runtime.Object.
 // It returns a pointer to a metav1.ObjectMeta structure if successful, along with any error encountered.
 func getObjectMetadataFromRuntimeObject(obj runtime.Object) (*metav1.ObjectMeta, error) {
+	fmt.Println(obj)
```

Comment:
Remove. And thanks for checking that this doesn't contain anything beyond the metadata!
```suggestion
```